frontend/src/components/GameDisplay.jsx

```diff
@@ -3,142 +3,30 @@ import * as THREE from "three";
 import { OrbitControls, PerspectiveCamera, Text } from '@react-three/drei';
 import { useFrame } from '@react-three/fiber';
 
-import { levels } from "../levels";
-import { BUILD, DEFEND, SCORE } from "../Game";
+import { BUILD } from "../Game";
 
 import MapView from "./views/MapView";
 import EnemyView from "./views/EnemyView";
 import StructureView from "./views/StructureView";
 import ProjectileView from "./views/ProjectileView";
-import GameInfo from "./ui/GameInfo";
-import BuyMenu from "./ui/BuyMenu";
-import SellMenu from "./ui/SellMenu";
 import PathView from "./views/PathView";
-import { Status as TowerStatus } from "../entities/towers/Tower";
+import SelectedTowerView from "./views/SelectedTowerView";
 
-export default function GameDisplay({game, assets}) {
-    const { gameMap, castle } = game;
+export default function GameDisplay({game, assets, selectedTower}) {
+    const { gameMap } = game;
     const { width, depth, height } = gameMap;
 
-    const [goldReward, setGoldReward] = useState(100);
-    const [enemies, setEnemies] = useState([]);
-
-    const [projectiles, setProjectiles] = useState([]);
+    const [ticks, setTicks] = useState(0);
 
     useFrame((_state, _delta, _xrFrame) => {
         game.tick();
 
-        handleEnemiesAtCastle(game.enemies);
-        towersAttack(game.enemies);
-
-        setEnemies([...game.enemies]);
-        setProjectiles([...game.projectiles]);
-
-        checkLevelOver();
+        setTicks(ticks + 1); // this is a hack to make the canvas re-render, open to improvements
```

Comment:
Could you include what the consequence are without the hack, i.e. what happens if the canvas doesn't re-render?
```suggestion
        setTicks(ticks + 1); // this is a hack to make the canvas re-render
```